tests/test_core/test_new_alignment.py

```diff
@@ -1100,32 +1101,29 @@ def test_sequence_collection_get_translation_error(data):
         seq_coll.get_translation()
 
 
-@pytest.mark.xfail(
-    reason="todo: relies on Sequence.get_translation which requires a codon alphabet class'"
-)
 @pytest.mark.parametrize(
     "data",
     (
+        {"seq1": "GATTTT", "seq2": "GATCTT"},
+        {"seq1": "GAT---", "seq2": "GGATCT"},
+        {"seq1": "GATTTT", "seq2": "?GATCT"},
         {"seq1": "GATTTT", "seq2": "GATC??"},
-        {"seq1": "GAT---", "seq2": "?GATCT"},
+        {"seq1": "GAT-T-", "seq2": "GATCTT"},
     ),
 )
 def test_sequence_collection_get_translation_info(data):
     """SequenceCollection.get_translation preserves info attribute"""
-    seq_coll = new_aln.make_unaligned_seqs(data, moltype="dna")
-    got = seq_coll.get_translation()
+    seq_coll = new_aln.make_unaligned_seqs(data, moltype="dna", info={"key": "value"})
+    got = seq_coll.get_translation(incomplete_ok=True)
     assert got.info["key"] == "value"
 
 
-@pytest.mark.xfail(
-    reason="todo: relies on Sequence.get_translation which requires a codon alphabet class'"
-)
 def test_sequence_collection_get_translation_incomplete():
     """get translation works on incomplete codons"""
     data = {"seq1": "GATN--", "seq2": "?GATCT"}
     seq_coll = new_aln.make_unaligned_seqs(data, moltype="dna")
     got = seq_coll.get_translation(incomplete_ok=True)
-    assert got.to_dict() == {"seq1": "D?", "seq2": "XS"}
+    assert got.to_dict() == {"seq1": "DX", "seq2": "XS"}
```

Comment:
hmmm, should we change behaviour so "X" becomes "?"?.

Add a comment to revisit this behaviour.